test/test_output_readers.jl

```diff
@@ -0,0 +1,129 @@
+using Test
+using JLD2
+
+using Oceananigans
+using Oceananigans.Units
+
+function generate_some_interesting_simulation_data(Nx, Ny, Nz; architecture=CPU())
+    grid = RegularRectilinearGrid(size=(Nx, Ny, Nz), extent=(64, 64, 32))
```

Comment:
Does this test need to be so big?

Reply:
Ah definitely not. I just wanted some interesting data that would produce unique data for u, v, w, T, and S when slices/averaged/etc. and instead of thinking too hard thought it would be easier to copy paste the example.

But now I don't think the tests will be that extensive.